### PR TITLE
Math Magician: Events

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this Pull Request, I: 

- [x] Copy the files `calculate.js` and `operate.js` into a `logic/` directory in my project.
- [x] Analyze the files `calculate.js` and `operate.js`.
- [x] Import the files in my `Calculator` component.
- [x] Use `state` and `props`, pass props to child components, use lifecycle methods, lift the state up.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,43 +1,78 @@
...
-          <button type="button" className="button-section operator">-</button>
-          <button type="button" className="button-section">1</button>
-          <button type="button" className="button-section">2</button>
-          <button type="button" className="button-section">3</button>
-          <button type="button" className="button-section operator">+</button>
+          <button type="button" className="button-section" value="AC" onClick={() => { this.calculator(digitObj, 'AC'); }}>AC</button>
+          <button type="button" className="button-section" value="+/-" onClick={() => { this.calculator(digitObj, '+/-'); }}>+/-</button>
+          <button type="button" className="button-section" value="%" onClick={() => { this.calculator(digitObj, '%'); }}>%</button>
+          <button type="button" className="button-section operator" value="+" onClick={() => { this.calculator(digitObj, '+'); }}>+</button>
+          <button type="button" className="button-section" value="7" onClick={() => { this.calculator(digitObj, '7'); }}>7</button>
```
Comment (DeliceLydia) [inline src/components/Calculator.js:49]: - [ ] You repeated `+` operator this one is supposed to work as a `division` please work on that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Ntwali-Josue, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (Ntwali-Josue) [reply src/components/Calculator.js:49]: Oh, I did not see this, thanks for pointing this out, I am going to work on it. Thanks, @DeliceLydia.

**Review by safafa — APPROVED** [review]
## STATUS: APPROVED 🟢 👏
Hello @Ntwali-Josue  great job working on the required changes 👍 💯

Now you can merge this branch 💪 😄
HAPPY CODING!! AND GOOD LUCK WITH THE NEXT ONES 🤞 🍀

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ntwali-Josue** [comment]
@safafa thanks a lot for approving my project

**Pull request merged**